crates/node/src/cli/rpc.rs

```diff
@@ -185,6 +181,16 @@ impl RpcParams {
         SocketAddr::new(listen_addr.into(), self.rpc_port)
     }
 
+    pub fn addr_admin(&self) -> SocketAddr {
+        let listen_addr = if self.rpc_external {
```

Comment:
a specific bind configuration is required for the admin interface
we may want to expose the normal rpc methods externally, while restricting the admin rpc methods to local access, even if they are on a different port

however, ensure that both normal and admin RPC methods remain accessible externally when running inside Docker

Reply:
Yep, I'm assuming this is the responsibility of the node operator. For example, it might be worthwhile opening the admin methods to 0.0.0.0, but with a proxy requiring an admin api token and ip whitelist.

Reply:
I think I see more clearly what you were talking about. I have updated the `rpc-methods` cli arg (which is now rpc-endpoints) to handle this:

- Auto: `admin` and `user` are exposed on `localhost` by default. If `rpc-external` is _true_, `user` is exposed on `0.0.0.0` but `admin` stays exposed on `localhost`.
- Safe: `admin` is disabled and is _not_ exposed, even if `rpc-external` is set to _false_.
- Unsafe: `admin` is exposed on `0.0.0.0` if `rpc-external` is set to true.